modvendor/modvendor_test.go

```diff
@@ -15,18 +15,823 @@
 package modvendor_test
 
 import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strings"
 	"testing"
+	"text/template"
 
 	"github.com/madlambda/spells/assert"
 	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
 	"github.com/mineiros-io/terramate/modvendor"
 	"github.com/mineiros-io/terramate/test"
+	errtest "github.com/mineiros-io/terramate/test/errors"
 	"github.com/mineiros-io/terramate/test/sandbox"
 	"github.com/mineiros-io/terramate/tf"
 	"github.com/rs/zerolog"
+
+	. "github.com/mineiros-io/terramate/test/hclwrite/hclutils"
 )
 
+// vendorPathSpec describes paths inside a vendor directory when its
+// directory is not yet know. It's format is just:
+//   <module source>#<vendor relative path>
+// Where the module source can use the "text/template" (dot) variable to be
+// substituted at a later time when the final vendor directory is computed.
+//
+// Example:
+//   git::file://{{.}}/module-test?ref=main#main.tf
+type vendorPathSpec string
+
+type testcase struct {
+	name         string
+	source       string
+	layout       []string
+	vendordir    string
+	configs      []hclconfig
+	wantVendored []string
+	wantIgnored  []wantIgnoredVendor
+	wantError    error
```

Comment:
hmm since Vendor returns a modvendor.Report, why we just dont define a want modvendor.Report instead of 3 different want fields ? :thinking:

Reply:
because it reads better in the table tests (IMO).. flat > nested in this case I guess. But not a big deal, can change back.

Reply:
hmm I can see your point...but I do get a sense of extra indirection that can cause some confusion. When it is just want and wantErr it is easier to understand, but in this case it is two slices that are related and will be composed in a final report...IDK :thinking: . Lets leave it like this, it is not bad, was just curious on the reasoning and I never felt annoyed by the nesting :laughing: .